### ISSUE TITLE
Clang-format discrepancy

### ISSUE THREAD

**Issue by supahero1** [cuberite/cuberite#5584]
Has clang-format ever been applied? I just applied it to the codebase and most of the files get changed by it (using the config file at the top of the repo). Mostly function arguments, some way-too-long comments, whitespace.

**Comment by bearbin** [comment]
The answer's no :(

Our style guide is enforced by the CheckBasicStyle lua script rather than clang-format. Switching over would be a good idea in the long term but probably not worth a big investment right at the moment.

**Comment by bearbin** [comment]
See https://github.com/cuberite/cuberite/tree/clang-format-codebase for what would be required to apply clang-format to the codebase. Probably a good idea, but the format might need some tweaking (& getting the PR backlog done would be nice to avoid conflicts!).

**Comment by MightyFilipns** [comment]
The clang-tidy.sh is broken in visual studio
Visual Studio does not support the generation of console_commands.json
The workaround is using the Ninja generator.
However, constantly switching between the two generators isn't practical because you have to delete the CMakeCache.txt file manually every time.
Also, I had problems with Lua not being found despite it being installed and I $PATH but I think that is because something else is broken on my machine.
Also, how is the run-clang-tidy command supposed to work? At first, I thought it was related to [this](https://github.com/lmapii/run-clang-tidy), but I figured out it's a Python script. How does the bash shell know that it is a Python script? What am I missing?

**Comment by bearbin** [comment]
Yep, that script is not intended for developer use really, if it works then great, otherwise you can work out how to run clang-tidy in your environment or just rely on the buildserver. `clang-tidy.sh` is really just for the buildserver to use. Visual Studio [should support clang-tidy natively](https://learn.microsoft.com/en-us/cpp/code-quality/clang-tidy?view=msvc-170) anyway, so if you want to use that IDE it should already be working.

The `run-clang-tidy` in that script is indeed the python version. On debian (and probably other linux distributions of clang-tidy) it's installed into `/usr/bin/run-clang-tidy` when you install clang-tidy. Your shell knows what interpreter to execute by the [hashbang](https://en.wikipedia.org/wiki/Shebang_(Unix)) at the beginning of the script.

**Comment by MightyFilipns** [comment]
Aha, I don't really use Linux much so I was confused

**Comment by MightyFilipns** [comment]
Is the new .clang-format supposed to format the code exactly as required by CheckBasicStyle.lua?
